src/main.ts

```diff
@@ -285,15 +285,18 @@ export default class SlidingPanesPlugin extends Plugin {
       // the amount of space to the right we need to leave for sticky headers
       const headersToRightWidth = this.settings.stackingEnabled ? (leafCount - this.activeLeafIndex - 1) * this.settings.headerWidth : 0;
 
+      // determine whether to request 'smooth' animations or 'auto' snap
+      let behavior: ScrollBehavior = animated && this.settings.smoothAnimation ? 'smooth' : 'auto';
+
       // it's too far left
       if (rootContainerEl.scrollLeft > position - left) {
         // scroll the left side of the pane into view
-        rootContainerEl.scrollTo({ left: position - left, top: 0, behavior: animated ? 'smooth': 'auto' });
+        rootContainerEl.scrollTo({ left: position - left, top: 0, behavior: behavior });
```

Comment:
There are some cases where `animated = false` is passed in (when closing panes or whatever), so to cater for these you should probably do `animated ? behavior : 'auto'`. Same on line 299.

Reply:
Hi, @deathau. Thanks for taking a look!

I believe I've handled that case with the logic on line 289:

```typescript
// determine whether to request 'smooth' animations or 'auto' snap
let behavior: ScrollBehavior = animated && this.settings.smoothAnimation ? 'smooth' : 'auto';
```

I.e., `behavior` isn't simply the user's setting but is derived from it. In the case where `animated = false` is passed in, the value of `behavior` will already be `auto` and I _believe_ will have the desired effect.

Thoughts? If there is something I'm missing, or if you would prefer that I tweak the above, please let me know!

Reply:
So you have. My bad for not reading properly 😳